kale/embed/factorization.py

```diff
@@ -258,3 +265,162 @@ def inverse_transform(self, x):
         x_rec = x_rec + self.mean_
 
         return x_rec
+
+
+class MIDA(BaseEstimator, TransformerMixin):
+    """Maximum independence domain adaptation
+    Args:
+        n_components (int): Number of components to keep.
+        penalty (str): "l2" or None. Penalty to use for the optimization problem. Defaults to None.
+        kernel (str): "linear", "rbf", or "poly". Kernel to use for MIDA. Defaults to "linear".
+        lambda_ (float): Hyperparameter of the domain covariate dependence. Defaults to 1.0.
+        mu (float): Hyperparameter of the l2 penalty. Defaults to 1.0.
+        eta (float): Hyperparameter of the label dependence. Defaults to 1.0.
+        augmentation (bool): Whether using covariates as augment features. Defaults to False.
+        kernel_params (dict or None): Parameters for the kernel. Defaults to None.
+
+    References:
+        [1] Yan, K., Kou, L. and Zhang, D., 2018. Learning domain-invariant subspace using domain features and
+            independence maximization. IEEE transactions on cybernetics, 48(1), pp.288-299.
+    """
+
+    def __init__(
+        self,
+        n_components,
+        penalty=None,
+        kernel="linear",
+        lambda_=1.0,
+        mu=1.0,
+        eta=1.0,
+        augmentation=False,
+        kernel_params=None,
+    ):
+        self.n_components = n_components
+        self.kernel = kernel
+        self.lambda_ = lambda_
+        if penalty is not None and penalty not in ["l2"]:
+            raise ValueError("Unsupport penalty type: {}".format(penalty))
+        else:
+            self.penalty = penalty
+        self.mu = mu
+        self.eta = eta
+        self.augmentation = augmentation
+        if kernel_params is None:
+            self.kernel_params = {}
+        else:
+            self.kernel_params = kernel_params
+        self._lb = LabelBinarizer(pos_label=1, neg_label=-1)
```

Comment:
lb: can you make it more readable? 
lb: pound https://en.wikipedia.org/wiki/Pound_(mass)

Reply:
Check all *lb* please